client/header/activity-panel/panels/stock/card.js

```diff
@@ -0,0 +1,152 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { BaseControl, Button } from '@wordpress/components';
+import { Component, Fragment } from '@wordpress/element';
+import { compose } from '@wordpress/compose';
+import { withDispatch } from '@wordpress/data';
+
+/**
+ * WooCommerce dependencies
+ */
+import { Link, ProductImage } from '@woocommerce/components';
+
+/**
+ * Internal dependencies
+ */
+import { ActivityCard } from '../../activity-card';
+
+class ProductStockCard extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
+			quantity: props.product.stock_quantity,
+			editing: false,
+		};
+
+		this.beginEdit = this.beginEdit.bind( this );
+		this.cancelEdit = this.cancelEdit.bind( this );
+		this.onQuantityChange = this.onQuantityChange.bind( this );
+		this.updateStock = this.updateStock.bind( this );
+	}
+
+	componentDidUpdate() {
+		this.quantityInput && this.quantityInput.focus();
+	}
+
+	beginEdit() {
+		this.setState( { editing: true } );
+	}
+
+	cancelEdit() {
+		this.setState( {
+			editing: false,
+			quantity: this.props.product.stock_quantity,
+		} );
+	}
+
+	onQuantityChange( quantity ) {
+		this.setState( { quantity } );
+	}
+
+	updateStock() {
+		const { product, updateItem } = this.props;
+
+		this.setState( { editing: false }, () => {
+			const data = {
+				stock_quantity: this.state.quantity,
+				type: product.type,
+				parent_id: product.parent_id,
+			};
+
+			updateItem( 'products', product.id, data );
+		} );
+	}
+
+	getActions() {
+		const { editing } = this.state;
+
+		if ( editing ) {
+			return [
+				<Button onClick={ this.updateStock } isPrimary>
+					{ __( 'Save', 'woocommerce-admin' ) }
+				</Button>,
+				<Button onClick={ this.cancelEdit }>{ __( 'Cancel', 'woocommerce-admin' ) }</Button>,
+			];
+		}
+
+		return [
+			<Button isDefault onClick={ this.beginEdit }>
+				{ __( 'Update stock', 'woocommerce-admin' ) }
+			</Button>,
+		];
+	}
+
+	getBody() {
+		const { editing, quantity } = this.state;
+
+		if ( editing ) {
+			return (
+				<Fragment>
+					<BaseControl className="woocommerce-stock-activity-card__edit-quantity">
+						<input
+							className="components-text-control__input"
+							type="number"
+							value={ quantity }
+							onChange={ this.onQuantityChange }
```

Comment:
I'm unable to enter numbers. Here is the error

<img width="1071" alt="Screen Shot 2019-04-05 at 3 19 12 PM" src="https://user-images.githubusercontent.com/1922453/55599889-4da78780-57b6-11e9-99fb-3a975ed53c08.png">